test/lambda.lisp

```diff
@@ -1,37 +1,244 @@
+      "test type of lambda term")
+  (is obj-equalp
+      so1
+      (lambda:type-of-term nil
+                           (lambda:term
+                            (lambda:annotated-term nil
+                                                   lambterm)))
+      "test type of annotated term for the lambda term"))
+
+(define-test app-test
+  :parent geb.lambda
+  (is obj-equalp
+      so1
+      (lambda:type-of-term nil (lambda:app lambterm pair-of-units-term))
+      "type of function application term")
+  (is obj-equalp
+      (so-hom-obj so1-prod so1)
+      (lambda:type-of-term nil
+                           (lambda:fun
+                            (lambda:annotated-term nil
+                                                   appterm)))
+      "test annotated fun term")
+  (is obj-equalp
+      so1-prod
+      (lambda:type-of-term nil
+                           (lambda:term
+                            (lambda:annotated-term nil
+                                                   appterm)))))
+
+(define-test index-tests
+  (is obj-equalp
+      (so-hom-obj so0 so1)
+      (lambda:type-of-term context-list (lambda:index 0)))
+  (is obj-equalp
+      so01-coprod
+      (lambda:type-of-term context-list (lambda:index 1)))
+  (is obj-equalp
+      so0
+      (lambda:type-of-term context-list (lambda:index 2)))
+  (is obj-equalp
+      so1
+      (lambda:type-of-term context-list (lambda:index 3))))
+
+
+(define-test absurd-index-test
+  (is obj-equalp
+      so0
+      (lambda:type-of-term context-list
+                           (lambda:term
+                            (lambda:annotated-term context-list
+                                                   (lambda:absurd so1 (lambda:index 2)))))))
+
+(define-test exp-hom-test
+  (is obj-equalp
+      (so-hom-obj (coprod so0 (so-hom-obj so1 so1)) (prod so1 (so-hom-obj so0 so1)))
+      (lambda:type-of-term (list so0 so1)
+                           (lambda:fun
+                            (lambda:annotated-term (list so0 so1)
+                                                   (lambda:app (lambda:lamb
+                                                                (coprod so0
+                                                                        (geb.lambda.main:fun-type (lambda:index 0)
+                                                                                                 so1))
+                                                                (pair unit-term
+                                                                      (lambda:lamb so1
+                                                                                   (lambda:lamb so0
+                                                                                                (lambda:index 0)))))
+                                                               (lambda:right so0
+                                                                             (lambda:lamb so1 unit-term))))))))
```

Comment:
Super long, if we export `fun-type`

then it should be rexported through `lambda`

```lisp
(geb.utils:muffle-package-variance
 (uiop:define-package #:geb.lambda
   (:documentation "A basic lambda calculus model")
   (:mix #:geb.lambda.spec #:geb.lambda.trans #:geb.common #:common-lisp)
   (:use-reexport #:geb.lambda.spec #:geb.lambda.main #:geb.lambda.trans)))
```

since it is, we can replace it